contentctl/objects/abstract_security_content_objects/detection_abstract.py

```diff
@@ -338,171 +356,209 @@ def serialize_model(self):
+                if isinstance(d, str) and self.name == d:
+                    new_detections.append(self)
+                    replaced = True
+                else:
+                    new_detections.append(d)
             if replaced is False:
-                raise ValueError(f"Error, failed to replace detection reference in Baseline '{baseline.name}' to detection '{self.name}'")             
+                raise ValueError(
+                    f"Error, failed to replace detection reference in Baseline '{baseline.name}' "
+                    f"to detection '{self.name}'"
+                )
             baseline.tags.detections = new_detections
 
         # Data source may be defined 1 on each line, OR they may be defined as
         # SOUCE_1 AND ANOTHERSOURCE AND A_THIRD_SOURCE
         # if more than 1 data source is required for a detection (for example, because it includes a join)
         # Parse and update the list to resolve individual names and remove potential duplicates
-        updated_data_source_names:set[str] = set()
-        
+        updated_data_source_names: set[str] = set()
+
         for ds in self.data_source:
             split_data_sources = {d.strip() for d in ds.split('AND')}
             updated_data_source_names.update(split_data_sources)
-        
+
         sources = sorted(list(updated_data_source_names))
-        
-        matched_data_sources:list[DataSource] = []
-        missing_sources:list[str] = []
+
+        matched_data_sources: list[DataSource] = []
+        missing_sources: list[str] = []
         for source in sources:
             try:
                 matched_data_sources += DataSource.mapNamesToSecurityContentObjects([source], director)
-            except Exception as data_source_mapping_exception:
+            except Exception:
                 # We gobble this up and add it to a global set so that we
                 # can print it ONCE at the end of the build of datasources.
                 # This will be removed later as per the note below
                 MISSING_SOURCES.add(source)
-                
+
         if len(missing_sources) > 0:
             # This will be changed to ValueError when we have a complete list of data sources
-            print(f"WARNING: The following exception occurred when mapping the data_source field to DataSource objects:{missing_sources}")
-        
+            print(
+                "WARNING: The following exception occurred when mapping the data_source field to "
+                f"DataSource objects:{missing_sources}"
+            )
+
         self.data_source_objects = matched_data_sources
 
         for story in self.tags.analytic_story:
-            story.detections.append(self)            
-        return self
+            # TODO (cmcginley): I think this instance of typing weirdness highlights some of the
+            #   issues I see with maintaining Detection and Detection_Abstract
+            story.detections.append(self)
```

Comment:
This is an instance of some of the weird typing situations we run into w/ `Detection` vs `Detection_Abstract`. `self` is of type `Detection_Abstract` but `story.detections` is declared as a list of `Detection` objects. This part of the `model_post_init` should be moved to the `Detection` class; or, better yet, we should remove the `Detection_Abstract` class altogether IMO.

Reply:
Resolving for now; I think theres a larger design convo to be had about `Detection` and `Detection_Abstract`